### DISCUSSION TITLE
what happens behind the curtain when...

### DISCUSSION THREAD

**Discussion by g13092** [mrlt8/docker-wyze-bridge#1234] [General]
When I'm viewing the webui and I click on the stale time in the upper right corner of a cam, it refreshes all the cams.  long stale snapshots are upated and the timers restart.  What is actually happening in the background when I click?

Why do i ask?

Occasionally, my feeds into frigate die, but coming back to the wyze-bridge ui, I can quickly restore them with a click on one of these timers.   I'd love to have HA ping the bridge now and again to keep things 'alive'.  Currently, I have  HA fire some MQTT commands that re-enable cams when it sees them as unavailable or an MQTT message that says they went offline.  I'm going to experiment with some others that might look for 0 FPS from the cams, then fire MQTT or HTTP commands.

**Comment by mrlt8** [comment]
Have you tried setting `ON_DEMAND=false`? The refresh snapshot option basically takes a snapshot from the rtsp stream and will start the stream if the bridge isn't connected to that camera.

**Reply by g13092** [reply]
I suppose I should have added that, yes, I do have `ON_DEMAND=false.`  I also have `CONNECT_TIMEOUT=25` and `IGNORE_OFFLINE=false`.  

This mainly happens to my more remote cameras, where the wifi isn't as strong.  It doesn't seem to take much to get them to come back when I click to get the snapshot to update.

**Reply by mrlt8** [reply]
hmm, can you post the `status` number next time it happens? You should be able to find it under the camera details on the web ui, if you click the little `i` icon on the top right of the camera preview.

**Reply by g13092** [reply]
absolutely.  I could also do /api and save the json if that would help.  I just looked now and noticed all cams were at 3.

**Reply by mrlt8** [reply]
just the number should be fine.

`3` means the camera is CONNECTED which is good.
These are the standard status numbers, but it will use the tutk error number (negative number) if there is an issue with the connection:

```python
    OFFLINE = -90
    STOPPING = -1
    DISABLED = 0
    STOPPED = 1
    CONNECTING = 2
    CONNECTED = 3
```

**Comment by g13092** [comment]
@mrlt8, looks like it's a 2.  I haven't caught as many issues in the past few days... I suppose I'll take that as a win.

**Reply by mrlt8** [reply]
`2` means the bridge is in the process of connecting to the camera. It should either connect `3` or time out after 20 seconds and go back to `1`. 

I wonder if accessing the web-ui triggered the re-connect? Could you try accessing the api endpoint directly to get the status?

**Comment by mrlt8** [comment]
@g13092 I was able to catch the camera getting stuck on `-1` most likely a blocking issue when the camera was stopping.

This seems related to #1240 issue, so let's continue this over there.
